### PR TITLE
cache deep_infra service request + add exec_profile parameter to pull…

### PR DESCRIPTION
**Pull request by zer0dss**
The bottleneck was similar to the previous one ( a request to DeepInfra API was executed for each Result object that was linked with DeepInfra ). 

I've also added the exec_profile parameter to the pull method. If present the method will print the exec time for downloading the JSON string of the object and the exec time to create an edsl object from it.

```python
from edsl import Results

r = Results.pull('f33a08a1-52c6-432a-9c05-e898ac76a2e2',exec_profile=True)
```

### CODE REVIEW

File: edsl/inference_services/DeepInfraService.py [hunk 1]
```diff
@@ -12,27 +12,34 @@ class DeepInfraService(InferenceServiceABC):
     _inference_service_ = "deep_infra"
     _env_key_name_ = "DEEP_INFRA_API_KEY"
 
+    _models_list_cache: List[str] = []
```
Comment (johnjosephhorton) [inline edsl/inference_services/DeepInfraService.py:15]: Is the idea that we'd list strings here as a fall-back?

**Review by johnjosephhorton — APPROVED** [review]
@zer0dss I like this - can we populate the class variable of models w/ the current deep infra snapshot?

**Comment by johnjosephhorton** [comment]
@zer0dss Looks like there's a tiny merge conflict - could you fix?

**Comment by zer0dss** [comment]
@johnjosephhorton I've fixed the conflict.

**Pull request merged**